### PR TITLE
[excel] (Events, Ranges) Adding section for data change detection

### PR DESCRIPTION
**Pull request by AlexJerabek**
Adding information about this upcoming feature. I'm putting the information in the range article, despite it being a table and worksheet event. I'm guessing this is a more intuitive place, but I'm open to feedback.

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[d35f4dd](https://github.com/OfficeDev/office-js-docs-pr/pull/873/commits/d35f4dd48c3249c8c529648453eef8d0a419a6ef)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-ranges-advanced.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-DataChange/docs/excel/excel-add-ins-ranges-advanced.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C4%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201904041911156089-873%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=MTSyDkoTtGv9F897jiKO0ExAzCoYXoGiAIJArGi0XW4%3D&st=2019-04-04T19%3A07%3A22Z&se=2019-05-05T19%3A12%3A22Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by kbrandl** [comment]
@AlexJerabek thanks for talking through this PR with me a few minutes ago. Let me know when you've updated this PR and it's ready for me to take another look. Thanks!

**Comment by AlexJerabek** [comment]
Thanks @kbrandl. The PR has been updated.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[06edec1](https://github.com/OfficeDev/office-js-docs-pr/pull/873/commits/06edec1d98773eb2a58d1695f03fd8618b263057)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-tables.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-DataChange/docs/excel/excel-add-ins-tables.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-worksheets.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-DataChange/docs/excel/excel-add-ins-worksheets.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C4%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201904042037488174-873%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=gTt1vIixA5L4IdiGREDCmUESX%2BXG0%2BjNjieoolG145A%3D&st=2019-04-04T20%3A33%3A55Z&se=2019-05-05T20%3A38%3A55Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/excel-add-ins-tables.md [hunk 1]
```diff
@@ -233,6 +232,30 @@ Excel.run(function (context) {
 
 ![Table data in Excel](../images/excel-tables-get-data.png)
 
+### Detect data changes
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-tables.md]: should be an H2 (instead of an H3)

File: docs/excel/excel-add-ins-tables.md [hunk 2]
```diff
@@ -233,6 +232,30 @@ Excel.run(function (context) {
...
+
+Your add-in may need to react to users changing the data in a table. To detect these changes, [Tables](/javascript/api/excel/excel.table) provide an `onChanged` event. This event fires whenever the data is changed in a particular table. Event handlers for `onChanged` receive a [TableChangedEventArgs](/javascript/api/excel/excel.tablechangedeventargs).
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-tables.md]: Suggest that you leave the first sentence as-is and then replace the other 3 sentences with the following:

To detect these changes, you can [register an event handler](excel-add-ins-events.md#register-an-event-handler) for the `onChanged` event of a [table](/javascript/api/excel/excel.table). Event handlers for the `onChanged` event receive a [TableChangedEventArgs](/javascript/api/excel/excel.tablechangedeventargs) object when the event fires.

(You can edit this comment in GitHub to copy/paste this markdown if you'd like to use it.)

File: docs/excel/excel-add-ins-worksheets.md [hunk 3]
```diff
@@ -274,6 +274,30 @@ Excel.run(function (context) {
 }).catch(errorHandlerFunction);
 ```
 
+## Detect data changes
+
+Your add-in may need to react to users changing the data in a range. To detect these changes, [Worksheets](/javascript/api/excel/excel.worksheet) provide an `onChanged` event. This event fires whenever the data is changed in a particular worksheet. Event handlers for `onChanged` receive a [WorksheetChangedEventArgs](/javascript/api/excel/excel.worksheetchangedeventargs).
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-worksheets.md]: First sentence should end with "worksheet" (not "range").

Then,suggest that you replace the 3 sentences after the opening sentence with the following:

To detect these changes, you can [register an event handler](excel-add-ins-events.md#register-an-event-handler) for the `onChanged` event of a [worksheet](/javascript/api/excel/excel.worksheet). Event handlers for the `onChanged` event receive a [WorksheetChangedEventArgs](/javascript/api/excel/excel.worksheetchangedeventargs) object when the event fires.

(You can edit this comment in GitHub to copy/paste this markdown if you'd like to use it.)

File: docs/excel/excel-add-ins-tables.md [hunk 4]
```diff
@@ -233,6 +232,30 @@ Excel.run(function (context) {
...
+
+The `TableChangedEventArgs` object provides information about the changes and the source. Since `onChanged` fires when either the format or value of the data changes, it can be useful to have your add-in check if the values have actually changed. The `details` property encapsulates this information as a [ChangedEventDetail](/javascript/api/excel/excel.changedeventdetail). The following code sample demonstrates how to display the before and after values and types of a cell that has been changed.
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-tables.md]: suggest replacing "demonstrates" with "shows"

File: docs/excel/excel-add-ins-worksheets.md [hunk 5]
```diff
@@ -274,6 +274,30 @@ Excel.run(function (context) {
...
+
+The `WorksheetChangedEventArgs` object provides information about the changes and the source. Since `onChanged` fires when either the format or value of the data changes, it can be useful to have your add-in check if the values have actually changed. The `details` property encapsulates this information as a [ChangedEventDetail](/javascript/api/excel/excel.changedeventdetail). The following code sample demonstrates how to display the before and after values and types of a cell that has been changed.
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-worksheets.md]: suggest replacing "demonstrates" with "shows"

**Comment by AlexJerabek** [comment]
Thanks @kbrandl. Updated with your suggestions.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[cddd791](https://github.com/OfficeDev/office-js-docs-pr/pull/873/commits/cddd791c28bc7455fc2cb499373ff67c06803866)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-tables.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-DataChange/docs/excel/excel-add-ins-tables.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-worksheets.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-DataChange/docs/excel/excel-add-ins-worksheets.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C4%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201904042154575251-873%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=7ufTN0Q9UNkFv5BSbK0AuRPlqsE82D7pdW%2BwHI4%2FXAk%3D&st=2019-04-04T21%3A51%3A08Z&se=2019-05-05T21%3A56%3A08Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**